bridge/ffi/src/c_api.rs

```diff
@@ -245,22 +230,18 @@ pub unsafe extern "C" fn graph_get_one_sided_private_friendship_connections(
 	GraphConnections { connections: connections_ptr, connections_len }
 }
 
-// Free GraphState
 #[no_mangle]
-pub unsafe extern "C" fn free_graph_state(graph_state: *mut GraphState) {
-	if GRAPH_STATES.is_none() {
-		return
-	}
-	let mut graph_states = GRAPH_STATES.as_ref().unwrap().lock().unwrap();
-	if let Some(index) = graph_states.iter().position(|&state| state == graph_state) {
-		let _ = Box::from_raw(graph_states.remove(index));
-	}
+pub extern "C" fn free_graph_state(graph_state: *mut GraphState) {
+	let graph_state_box = unsafe { Box::from_raw(graph_state) };
+	let mut graph_states = GRAPH_STATES.lock().unwrap();
+	graph_states.retain(|x| !std::ptr::eq(x, &graph_state_box));
+	drop(graph_state_box);
```

Comment:
This might be a double-deletion:
1. `graph_states.retain()` will drop the `GraphState` struct matching the pointer
2. Then we drop the pointer again explicitly.

Suggest:
* Eliminate `Box::from_raw()`; just use the pointer to call `graph_states.retain()` and the GraphState should be dropped.
* Might want to check that the graph was actually dropped (compare before & after count of `graph_states`... if this was the incorrect API instance, the graph would not be dropped; THEN we might want to drop it explicitly (or return error/false?)

Reply:
`graph_states.retain(|x| !std::ptr::eq(x.as_ref(), unsafe { &*graph_state })); `
drops it too as well from vector